boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay

```diff
@@ -27,6 +27,26 @@
 	aliases {
 		rtc = &pcf8563_xiao_round_display;
 	};
+
+	xiao_round_display_mipi_dbi {
+		compatible = "zephyr,mipi-dbi-spi";
+		spi-dev = <&xiao_spi>;
+		dc-gpios = <&xiao_d 3 GPIO_ACTIVE_HIGH>;
+		write-only;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		gc9a01_xiao_round_display: gc9a01@0 {
```

Comment:
Why moving this in the device tree ? This is more difficult to review and identify the changes ....

Reply:
This display uses a 4 wire SPI interface for commands and data, which means it can be supported by the MIPI DBI API. For more information on this API, you can look at the initial RFC: #64587. The main idea of the API is to better support display hardware that explicitly accelerates SPI 3/4 wire mode displays. This change will allow gc9x01 displays to operate with LCD hardware using the MIPI DBI API, as well as existing SPI controllers

For cases where a SPI peripheral is used to implement the 3/4 wire mode support, we need to use an emulated MIPI DBI driver that uses the SPI API. This driver is implemented by the `zephyr,mipi-dbi-spi` compatible, so we need to declare the display as a child of this driver's node.

Hope this helps explain the rationale for this change. Would you be willing to test it?